### DISCUSSION TITLE
Quart blueprints

### DISCUSSION THREAD

**Discussion by ccall48** [Breq16/flask-discord-interactions#56] [Q&A] [answered]
I am having a little trouble trying to get the blueprints working with Quart implementation. Are you able to add an example to your examples for using quart? I currently have it all in one file which works great, but its started to get to crowded and needs to get broken up.

My docker container shows following error when I try with current examples.
```
app_1  | Traceback (most recent call last):
app_1  |   File "/usr/src/app/app.py", line 27, in <module>
app_1  |     discord.register_blueprint(example_bp)
app_1  |   File "/usr/local/lib/python3.9/site-packages/flask_discord_interactions/discord.py", line 413, in register_blueprint
app_1  |     app.discord_commands.update(blueprint.discord_commands)
app_1  | AttributeError: 'function' object has no attribute 'discord_commands'
```

**Comment by breqdev** [comment, marked as answer]
Using Quart vs Flask shouldn't impact the blueprint registration part of things--pretty much the only difference internally between how this library handles Flask and Quart is inside the call to `DiscordInteractions.set_route`, so declaring blueprints in Flask vs Quart *should* be exactly the same. When I get a chance, I'll write up an example to verify this.

From the error you posted, it seems like you're calling `discord.register_blueprint()` with a function instead of a `DiscordInteractionsBlueprint` object. Regardless of if you're using Flask or Quart, you need to define your blueprint as a `DiscordInteractionsBlueprint`, not a `Flask.Blueprint` or a `Quart.Blueprint`.

> Similarly to Flask, you can use Blueprints to split your bot’s commands across multiple files. You can’t use Flask Blueprints for this, because all interactions from Discord are sent to the same URL. However, this library provides a DiscordInteractionsBlueprint class which you can use similarly to a Flask blueprint.
>
> https://flask-discord-interactions.readthedocs.io/en/latest/discord.html#blueprints

When I get a chance, I'll write up a better example of blueprint usage with Quart. I agree that it would be good to have!

**Reply by ccall48** [reply]
Thanks for taking the time to respond. I see my mistake now, i was very silly boi and didn't notice that i also named the function the same name as the DiscordInteractionsBlueprint 🤦‍♂️.
